Rent a new linter

rafaellehmkuhl opened the pull request:
Uses vue/recommended + vue/airbnb rules on Eslint

I did the necessary changes on the main frontend files to comply to the new rules.
There are several changes needed on the old/dev frontend that I still need to make, but I'm already opening the PR so you can review the structure.

patrickelectric reviewed (changes requested):
@rafaellehmkuhl it appears that CI is failing.

rafaellehmkuhl commented:
> @rafaellehmkuhl it appears that CI is failing.

Yes. It's failing on the development frontend. I'm finishing the fix on those.

rafaellehmkuhl commented:
@patrickelectric just finished those. I will test the frontends to see if nothing was broken during the lint fixes.

rafaellehmkuhl commented:
Just tested the frontends. There was a small bug I had introduced on the versionchooser interface, making it not show the entire repo name up, but its now fixed.

rafaellehmkuhl commented on core/services/versionchooser/frontend/index.html:
```diff
@@ -156,10 +156,18 @@ <h3>Applying...</h3>
+    new Vue({ // eslint-disable-line no-new
       el: '#app',
-      data () {
+      filters: {
```
[This one](https://eslint.vuejs.org/rules/order-in-components.html)
